selectors-4/deferred-for-level-5

```diff
@@ -127,3 +127,14 @@ The local link pseudo-class '':local-link''</h3>
 		What about if the document url has less than N segments,
 		and the link is same-page?
 		Should "null segments" count as matching, or not?
+
+<h3 id="custom-pseudo">Exposing custom state: the '':state()'' pseudo-class</h3>
+
+  The <dfn selector>:state()</dfn> pseudo-class takes an argument and matches
+  custom elements whose exposed custom states include the argument.
+
+  The grammar of the '':state()'' pseudo-class is:
+
+  <pre class=prod>:state( <ident> )</pre>
```

Comment:
Existing WPTs & impls make this case sensitive so:

```suggestion
  <pre class=prod>:state( <custom-ident> )</pre>
```

Reply:
I suspect we don't want the exclusion of CSS-wide keywords that `<custom-ident>` implies; I think we have at least some amount of consensus that such exclusion was a bad idea in the first place and places that needed it should be using `<dashed-ident>` instead, and that `<custom-ident>` was a mistake.  But on the other hand we should clearly document the case-sensitivity.  I'm not sure what the current best practice is.

(I can't find any reference for this longer than https://github.com/w3c/csswg-drafts/issues/7431#issuecomment-1276732615 but there might be a more detailed explanation somewhere.)

Reply:
so should we do `<dashed-ident>` for now...? or keep it as `<ident>`?

Reply:
`<dashed-ident>` is definitely wrong as this can take any ident. I suppose `<ident>` is right from a strictly grammar standpoint because, as @dbaron points out, it doesn’t exclude css wide keywords. `<ident>` claims case insensitivity though. Perhaps we could make a new token type here like `<cased-ident>`. Otherwise we should keep it as `<ident>` and either explain (or remove) the implementation handles cases discretely.

Reply:
I'm not sure what is best here. @tabatkins can you advise?

Reply:
> `<ident>` claims case insensitivity though.

Where are you seeing that? If there's any such statement it's an error and I can fix it; <https://drafts.csswg.org/css-values/#textual-values> defines `<ident>` as the union of keywords and author-defined identifiers, and since the latter are case-sensitive the `<ident>` must be too, by default.

Anyway, yeah, `<ident>` is the correct production to use here if the value set is *not* dashed and the CSS-wide keyword restriction isn't appropriate (both of which are true here).

Reply:
I can’t find the reference to case insensitivity now. 

The string given to customstateset can be dashed so should this be `<ident> | <dashed-ident>`?

Reply:
No, `<dashed-ident>` is a subset of `<ident>`, so just `<ident>` is sufficient.

Reply:
Ok so since i'm currently using `<ident>` can we merge this?

Reply:
I think you should add a sentence saying it's case-sensitive.  (unless @tabatkins disagrees, at least.)

Reply:
I amended the first sentence to say that the argument is case sensitive